compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionUtils.cpp

```diff
@@ -56,6 +63,59 @@ static bool canInsOperandTieWithOutsOperand(OpOperand *insOperand) {
   return llvm::any_of(linalgOp.getOutputOperands(), canTieWithOutsOperand);
 }
 
+/// Checks if a linalg op is a simple reduction of the innermost dimensions
+/// with identity map for the input.
+static bool isSimpleReduction(linalg::LinalgOp linalgOp) {
```

Comment:
That would match things like matmul op? Is this what we want? I guess we could refine the heuristic later.

Reply:
No, it actually uses the matchReduction utility function, so matmul won't be matched.